app/models/city.rb

```diff
@@ -0,0 +1,9 @@
+class City < ActiveRecord::Base
+  validates :name, presence: true
+  has_many :stations
+
+  def self.record(city_information)
+    self.find_or_create_by(city_information)
```

Comment:
this find_or_create_by method is awesome. it checks to see if there is a record that already has matching information, and if it doesn't it will create a new record. basically, it combines the .create and .find_by methods.